ladybug_geometry/geometry3d/sphere.py

```diff
@@ -0,0 +1,158 @@
+        radius: A number representing the radius of the arc.
+
+    Properties:
+        * center
+        * radius
+        * min
+        * max
+        * diameter
+        * circumference
+        * area
+        * volume
+    """
+    __slots__ = ('_center', '_radius')
+
+    def __init__(self, center, radius):
+        """Initilize Sphere3D.
+        """
+        assert isinstance(center, Point3D), \
+            "Expected Point3D. Got {}.".format(type(center))
+        assert radius > 0, 'Sphere radius must be greater than 0. Got {}.'.format(radius)
+        self._center = center
+        self._radius = radius
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a Sphere3D from a dictionary.
+
+        Args:
+            data: A python dictionary in the following format
+
+        .. code-block:: python
+
+            {
+            "type": "Sphere3D"
+            "center": (10, 0, 0),
+            "radius": 5,
+            }
+        """
+        return cls(Point3D.from_array(data['center']), data['radius'])
+
+    @property
+    def center(self):
+        """Center of sphere."""
+        return self._center
+
+    @property
+    def radius(self):
+        """Radius of sphere."""
+        return self._radius
+
+    @property
+    def min(self):
+        """A Point3D for the minimum bounding rectangle vertex around this geometry."""
+        return Point3D(self.center.x - self.radius, self.center.y - self.radius,
+                       self.center.z - self.radius)
+
+    @property
+    def max(self):
+        """A Point3D for the maximum bounding rectangle vertex around this geometry."""
+        return Point3D(self.center.x + self.radius, self.center.y + self.radius,
+                       self.center.z + self.radius)
+
+    @property
+    def diameter(self):
+        """Diameter of sphere"""
+        return self.radius * 2
+
+    @property
+    def circumference(self):
+        """Circumference of sphere"""
+        return 2 * math.pi * self.radius
+
+    @property
+    def area(self):
+        """Area of sphere"""
+        return 4 * math.pi * self.radius ** 2
+
+    @property
+    def volume(self):
+        """Volume of sphere"""
+        return 4 / 3 * math.pi * self.radius ** 3
+
+    def move(self, moving_vec):
+        """Get a sphere that has been moved along a vector.
+
+        Args:
+            moving_vec: A Vector3D with the direction and distance to move the shere.
+        """
+        return Sphere3D(self.center.move(moving_vec), self.radius)
+
+    def scale(self, factor, origin=None):
+        """Scale a arc by a factor from an origin point.
+
+        Args:
+            factor: A number representing how much the sphere should be scaled.
+            origin: A Point3D representing the origin from which to scale.
+                If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        return Sphere3D(self.center.scale(factor, origin), self.radius * factor)
+
```

Comment:
It would be good if you could include methods for `rotate` and `reflect` like the other geometry objects.  I know that those two don't get used as much as the `move` and `scale` methods do but they have their usefulness and I am trying to ensure that all geometry objects support all 4 of the [similar transformations](https://en.wikipedia.org/wiki/Similarity_(geometry)):
* move
* scale
* rotate
* reflect

Reply:
My bad!  With the sphere not having an orientation and wrongly assuming these where local transformations.  Just needed to see the inputs, no excuse!

I added the rotate_xy since I see that is implemented in other classes as well. Hope that's okay.